spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationGraphServer.java

```diff
@@ -162,6 +164,19 @@ private void gateways(Collection<IntegrationNode> nodes, Collection<LinkNode> li
 			nodes.add(gatewayNode);
 			producerLink(links, channelNodes, gatewayNode);
 		}
+		Map<String, GatewayProxyFactoryBean> gpfbs = this.applicationContext
+				.getBeansOfType(GatewayProxyFactoryBean.class);
+		for (Entry<String, GatewayProxyFactoryBean> entry : gpfbs.entrySet()) {
+			Map<Method, MessagingGatewaySupport> methodMap = entry.getValue().getGateways();
+			int i = 0;
+			for (Entry<Method, MessagingGatewaySupport> gwEntry : methodMap.entrySet()) {
+				MessagingGatewaySupport gateway = gwEntry.getValue();
+				MessageGatewayNode gatewayNode = this.nodeFactory.gatewayNode(
+						entry.getKey().substring(1) + "." + gwEntry.getKey().getName() + "#" + i++, gateway);
```

Comment:
I quite like the longer names Artem suggests. What hash+index means isn't totally obvious to me but a full signature is.  If some tool in the future was offering a 'open node in IDE' then having the full signature would be nice.